### ISSUE TITLE
Argument types are not respected in JSDoc

### ISSUE THREAD

**Issue by trezm** [trezm/type-doc#41]
This function call should yield an error:

```
/**
 * Really doesn't do much...
 *
 * @param {Number} someNumber - a number
 * @returns {String} the string of a number
 */
function someFunction(someNumber) {
    const someString /* t:String */ = someNumber;
    return someString;
}
```
but does not, whereas this function, which should be equivalent, does
```
function someFunction(someNumber /* t:Number */) {
    const someString /* t:String */ = someNumber;
    return someString;
}
```

**Comment by trezm** [comment]
See [runkit](https://runkit.com/pete/typedoc-pr-41)

**Comment by trezm** [comment]
See [runkit](https://runkit.com/pete/typedoc-pr-41/2.0.0) for working published version

**Comment by trezm** [comment]
Resolved with #42
